arcade/sprite/base.py

```diff
@@ -306,16 +309,15 @@ def visible(self) -> bool:
             sprite.visible = not sprite.visible
 
         """
-        return self._color[3] > 0
+        return self._visible
 
     @visible.setter
     def visible(self, value: bool):
-        self._color = Color(
-            self._color[0],
-            self._color[1],
-            self._color[2],
-            255 if value else 0,
-        )
+        if self._visible == value:
```

Comment:
If ``value`` is an int, this version will store it instead of a bool. I'm not sure what the best approach here is. The shortest would be as follows:
```suggestion
        if (value := bool(value)) == self._visible:
```
You could also split out the assignment into two lines:
```py
value = bool(value)
if self._visible == value:
```